### DISCUSSION TITLE
Help with quarto-action (CI) on Branching/Pull Requests without overwriting the main GitHub Pages

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#3027] [Q&A]
As a software developer, I'm used to branching when an important change is made to the project (rather than modifying the main branch). The CI Action in this case serves as a way to validate the changes before they go "live". 

However, with my current action setup at https://github.com/fuhrmanator/log210-ndc-quarto/blob/main/.github/workflows/quarto-render-publish.yml when I create a Pull Request, the GitHub pages ("live result") is overwritten. 

I'm assuming there is there a way to put the "branch" in the path of the published results (so the "main" result sent to GitHub pages isn't overwritten)?

**Comment by mcanouil** [comment]
For R Lille (R User Group in Lille, France), I am using Netlify preview build on PR, but the idea for GitHub Pages would be the same without the preview.
I have the following workflow which allow me to manually trigger it which will install the latest Quarto, render the website and make a PR.
Then Netlify will "build" and deploy the website for me to see it live before merging the PR.
<https://github.com/RLille/rlille.fr/blob/main/.github/workflows/build.yml>

For instance, I just triggered the action resulting in the following:
- PR: https://github.com/RLille/rlille.fr/pull/49
- Netlify Preview: https://deploy-preview-49--rlille.netlify.app/

**Reply by fuhrmanator** [reply]
Thanks - I will check out Netlify. It seems it's not possible with GitHub Pages: https://stackoverflow.com/a/33097953/1168342

**Reply by fuhrmanator** [reply]
I found some hints at https://github.com/actions/deploy-pages/blob/69a62cdfa4c32e0d925268597986261876cac577/action.yml#L36

Perhaps this will be possible in the future with GitHub Pages.

**Reply by mcanouil** [reply]
You can't deploy multiple branches with gh-pages.
But you can get a PR with your changes that you can merge to gh-pages

**Reply by fuhrmanator** [reply]
> But you can get a PR with your changes that you can merge to gh-pages

Where can you "preview" it, as you do with your https://deploy-preview-49--rlille.netlify.app/ ? Are you saying to use Netlify for PRs and GH pages for the main build?

**Reply by mcanouil** [reply]
The workflow is the same for both Netlify (using `main`) as with GitHub Pages (using `gh-pages`), i.e., you can make changes, render, make a Pull Request.
After that, GitHub only allow one branch (or eventually directory) to be deployed, while Netlify will deploy main branch and PRs branches.
So if you care about previewing the HTML files, then you have no other choices than to use a workaround (i.e., something like the 'dev' website for `pkgdown`) or switch to a web service such as Netlify, at least far as I know.

**Comment by burgerga** [comment]
It's a couple of years late, but hopefully this will help someone else: you can use https://github.com/marketplace/actions/deploy-pr-preview .

Apart from setting up the preview, you need to change the main gh-pages workflow from [quarto-dev/quarto-actions/publish](https://github.com/quarto-dev/quarto-actions/blob/main/publish) to [quarto-dev/quarto-actions/render](https://github.com/quarto-dev/quarto-actions/blob/main/render), because the publish action force pushes the results to the gh-pages branch, wiping out any PR previews. Instead, you can deploy using 

```
      - name: Deploy 🚀
        uses: JamesIves/github-pages-deploy-action@v4
        with:
          folder: _site/
          clean-exclude: pr-preview/
          force: false
```

You can find a full example at https://github.com/gofair-foundation/gofair-wiki/tree/b09b6ddca82d9126522294710cd7aca8ebcb8326/.github/workflows .
